lib/modules/datasource/rpm/index.ts

```diff
@@ -0,0 +1,219 @@
+    }
+  }
+
+  // Fetches the filelists.xml.gz URL from the repomd.xml file.
+  @cache({
+    namespace: `datasource-${RpmDatasource.id}`,
+    key: (registryUrl: string) => registryUrl,
+    ttlMinutes: 1440,
+  })
+  async getFilelistsGzipUrl(registryUrl: string): Promise<string | null> {
+    const repomdUrl = joinUrlParts(
+      registryUrl,
+      RpmDatasource.repomdXmlFileName,
+    );
+    const response = await this.http.getText(repomdUrl.toString());
+
+    // check if repomd.xml is in XML format
+    if (!response.body.startsWith('<?xml')) {
+      logger.debug(
+        { datasource: RpmDatasource.id, url: repomdUrl },
+        'Invalid response format',
+      );
+      throw new Error(
+        `${repomdUrl} is not in XML format. Response body: ${response.body}`,
+      );
+    }
+
+    // parse repomd.xml using XmlDocument
+    const xml = new XmlDocument(response.body);
+
+    const filelistsData = xml.childWithAttribute('type', 'filelists');
+
+    if (!filelistsData) {
+      logger.debug(
+        `No filelists data found in ${repomdUrl}, xml contents: ${response.body}`,
+      );
+      throw new Error(`No filelists data found in ${repomdUrl}`);
+    }
+
+    const locationElement = filelistsData.childNamed('location');
+    if (!locationElement) {
+      throw new Error(`No location element found in ${repomdUrl}`);
+    }
+    const href = locationElement.attr.href;
+    if (!href) {
+      throw new Error(`No href found in ${repomdUrl}`);
+    }
+    // replace trailing "repodata/" from registryUrl, if it exists, with a "/" because href includes "repodata/"
+    const registryUrlWithoutRepodata = registryUrl.replace(
+      /\/repodata\/?$/,
+      '/',
+    );
+    return joinUrlParts(registryUrlWithoutRepodata, href);
+  }
+
+  async getReleasesByPackageName(
+    filelistsGzipUrl: string,
+    packageName: string,
+  ): Promise<ReleaseResult | null> {
+    let response: HttpResponse<Buffer>;
+    let decompressedBuffer: Buffer;
+    try {
+      // filelistsXmlUrl is a .gz file, need to extract it before parsing
+      response = await this.http.getBuffer(filelistsGzipUrl);
+      if (response.body.length === 0) {
+        logger.debug(`Empty response body from getting ${filelistsGzipUrl}.`);
+        throw new Error(
+          `Empty response body from getting ${filelistsGzipUrl}.`,
+        );
+      }
+      // decompress the gzipped file
+      decompressedBuffer = await gunzipAsync(response.body);
+    } catch (err) {
+      logger.debug(
+        `Failed to fetch or decompress ${filelistsGzipUrl}: ${
+          err instanceof Error ? err.message : err
+        }`,
+      );
+      throw err;
+    }
+
+    // Use XML streaming parser to handle large XML files efficiently
+    // If the file is too large (e.g., > 512MB), node.js doesn't support it natively.
+    // Therefore, we use a streaming parser to handle the XML file.
+    // This allows us to parse the XML file without loading the entire file into memory.
```

Comment:
I had to parse XML streaming instead of the XML file itself because in a real case the decompressed XML is larger than 512MB and node.js doesn't support a string larger than that.